src/wallet/mod.rs

```diff
@@ -805,9 +805,14 @@ where
             }
         };
 
-        fee_amount += fee_rate.fee_vb(serialize(&drain_output).len());
-
-        let drain_val = (coin_selection.selected_amount() - outgoing).saturating_sub(fee_amount);
+        // sum(inputs) - sum(outputs)
+        let delta = coin_selection.selected_amount() - outgoing;
+        // Fee to be paid if we add the drain_output
+        let drain_fee = fee_rate.fee_vb(serialize(&drain_output).len());
```

Comment:
I'm pretty sure that the old code was cumbersome, but correct, and this is just an equivalent way to write it.

Using `drain_fee` as `fee_rate.fee_vb(serialize(&drain_output).len())`, 
in the old code `drain_fee` was added to `fee_amount` (l808). So, `drain_val`
became the effective remaining value of the selection (l810).

If `drain_val` was dust, its value was added to `fee_amount` (l833), effectively computing
the following:
```
fee_amount = change_fee + drain_val + fee_amount
```
If it wasn't, the effective value `drain_val` was sent, as `change_fee`
was already considered inside `fee_amount`.

Note that:
```
delta = selected_amount - outgoing = drain_fee + drain_val + fee_amount
```

Reply:
Yeah, you're probably right. I think the main bug was just adding the drain_value to the fee_amount even when there was no drain output.

But I had to modify those lines anyways, so I just cleaned them up a bit ✨ 🧹